neuraxle/steps/flow.py

```diff
@@ -0,0 +1,199 @@
+        return self, data_container
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext) -> (
+            'BaseStep', DataContainer):
+        """
+        Nullify wrapped step hyperparams, and don't fit_transform the wrapped step.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: step, data_container
+        :type: (BaseStep, DataContainer)
+        """
+        self._nullify_hyperparams()
+        return self, data_container
+
+    def handle_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Nullify wrapped step hyperparams, and don't transform the wrapped step.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: step, data_container
+        :type: DataContainer
+        """
+        self._nullify_hyperparams()
+        return data_container
+
+    def _nullify_hyperparams(self):
+        """
+        Nullify wrapped step hyperparams using hyperparams_space.nullify().
+        """
+        hyperparams_space = self.wrapped.get_hyperparams_space()
+        self.wrapped.set_hyperparams(hyperparams_space.nullify())
+
+
+CHOOSE_ONE_OR_MANY_STEPS_OF_CHOICE_HYPERPARAM = 'choice'
+
+
+class ChooseOneOrManyStepsOf(Pipeline):
+    """
+    A pipeline to allow choosing many steps using an hyperparameter.
+
+    Example usage :
+
+    .. code-block:: python
+
+        p = Pipeline([
+            ChooseOneOrManyStepsOf([
+                ('a', Identity()),
+                ('b', Identity())
+            ])
+        ])
+        p.set_hyperparams({
+            'ChooseOneOrManyStepsOf': {
+                'choice': {
+                    'a': True
+                    'b': False
+                }
+            }
+        })
+
+    .. seealso::
+        :class:`Pipeline`
+        :class:`Nullify`
+    """
+
+    def __init__(self, steps, hyperparams=None):
+        Pipeline.__init__(self, steps)
+
+        if hyperparams is None:
+            hyperparams = HyperparameterSamples({
+                'choice': {}
+            })
+
+        self.set_hyperparams(hyperparams)
+
+    def set_hyperparams(self, hyperparams: Union[HyperparameterSamples, OrderedDict, dict]) -> BaseStep:
+        """
+        Set hyperparams for step selection, and nullify the steps that are not chosen.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return: self
+        :rtype: BaseStep
+        """
+        super().set_hyperparams(hyperparams=hyperparams)
+
+        if CHOOSE_ONE_OR_MANY_STEPS_OF_CHOICE_HYPERPARAM not in self.hyperparams:
+            raise ValueError('\'choice\' hyperparam not set in {0} hyperparams'.format(self.name))
```

Comment:
Remove this if. To remove it, it involves doing this: by default, all optionals / nullifiables will be to true. WOuld it work?